libbeat/processors/add_remote_metadata/add_remote_metadata_test.go

```diff
@@ -0,0 +1,207 @@
+	},
+	{
+		name: "invalid_no_match_keys",
+		config: mapstr.M{
+			"provider":       "map",
+			"target":         "new_field",
+			"include_fields": []string{"remote.key"},
+		},
+		wantInitErr: errors.New("fail to unpack the add_remote_metadata configuration: missing required field accessing 'match_keys'"),
+	},
+	{
+		name: "invalid_no_fields_or_target",
+		config: mapstr.M{
+			"provider":   "map",
+			"match_keys": []string{"remote.user.id"},
+		},
+		wantInitErr: errors.New("fail to unpack the add_remote_metadata configuration: no target field and no source fields specified accessing config"),
+	},
+	{
+		name: "valid_scalar",
+		config: mapstr.M{
+			"match_keys": []string{"remote.user.id"},
+			"target":     "new_field",
+			"provider":   "map",
+			"metadata": mapstr.M{
+				"one": "the value",
+				"two": "not the value",
+			},
+		},
+		event: mapstr.M{
+			"remote": mapstr.M{
+				"user": mapstr.M{
+					"id": "one",
+				},
+			},
+		},
+		want: mapstr.M{
+			"new_field": "the value",
+			"remote": mapstr.M{
+				"user": mapstr.M{
+					"id": "one",
+				},
+			},
+		},
+	},
+	{
+		name: "valid_object_root",
+		config: mapstr.M{
+			"match_keys":     []string{"remote.user.id"},
+			"include_fields": []string{"remote.key"},
+			"provider":       "map",
+			"metadata": mapstr.M{
+				"one": mapstr.M{"remote.key": "the value"},
+				"two": mapstr.M{"remote.key": "not the value"},
+			},
+		},
+		event: mapstr.M{
+			"remote": mapstr.M{
+				"user": mapstr.M{
+					"id": "one",
+				},
+			},
+		},
+		want: mapstr.M{
+			"remote": mapstr.M{
+				"key": "the value",
+				"user": mapstr.M{
+					"id": "one",
+				},
+			},
+		},
+	},
+	{
+		name: "valid_object_non_root",
+		config: mapstr.M{
+			"match_keys":     []string{"remote.user.id"},
+			"target":         "new_field",
+			"include_fields": []string{"remote.key"},
+			"provider":       "map",
+			"metadata": mapstr.M{
+				"one": mapstr.M{"remote.key": "the value"},
+				"two": mapstr.M{"remote.key": "not the value"},
+			},
+		},
+		event: mapstr.M{
+			"remote": mapstr.M{
+				"user": mapstr.M{
+					"id": "one",
+				},
+			},
+		},
+		want: mapstr.M{
+			"new_field": mapstr.M{
+				"remote": mapstr.M{
+					"key": "the value",
+				},
+			},
+			"remote": mapstr.M{
+				"user": mapstr.M{
+					"id": "one",
+				},
+			},
+		},
+	},
+	{
+		name: "doc_people",
+		config: mapstr.M{
+			"match_keys":     []string{"user.name"},
+			"include_fields": []string{"roles"},
+			"target":         "user",
+			"provider":       "map",
+			"metadata": mapstr.M{
+				"Alice": mapstr.M{"roles": []string{"ceo", "admin"}},
+				"Bob":   mapstr.M{"roles": []string{"hr"}},
+			},
+		},
+		event: mapstr.M{
+			"user": mapstr.M{
+				"name": "Alice",
+			},
+		},
+		want: mapstr.M{
+			"user": mapstr.M{
+				"name":  "Alice",
+				"roles": []any{"ceo", "admin"},
+			},
+		},
+	},
+}
+
+func TestAddRemoteMetadata(t *testing.T) {
+	logp.TestingSetup(logp.WithSelectors(name))
```

Comment:
Please fix the linter issue here.

Reply:
IMO this should be fixed in the linter settings; in the code base there a five instances of use of `logp.TestingSetup` that actually use the returned error. The vast majority of the remaining 233 do nothing and a minority add source noise by adding the `_ =` prefix needed to silence the linter.

Reply:
If you think that checking the error and failing the test because of it is not important just put the `_ =` prefix.

I don't think we need to change the linter configuration.

Reply:
Please take a look at the commit message for #36474.